addons/crm_iap_lead_enrich/models/crm_iap_lead_enrich.py

```diff
@@ -0,0 +1,150 @@
+            })
+        return log_data
+
+    @api.model
+    def _send_message(self, company_data=False):
+        if company_data:
+            messages_to_post = self._create_message_data(company_data)
+            self.message_post_with_view('crm_iap_lead.lead_message_template', values=messages_to_post, subtype_id=self.env.ref('mail.mt_note').id)
+        else:
+            self.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_not_found', subtype_id=self.env.ref('mail.mt_note').id)
+    @api.model
+    def _enrich_leads_from_response(self, data_clearbit):
+        """ This method will get the response from the service and enrich the lead accordingly """
+        for lead_id, data in data_clearbit.items():
+            record = self.browse(int(lead_id))
+            if record.exists():
+                if data:
+                    street = False
+                    zip_code = False
+                    city = False
+                    country_id = False
+                    state_id = False
+
+                    if not(record.street or record.street2 or record.zip or record.city or record.state_id or record.country_id):
+                        street = data["location"]
+                        zip_code = data["postal_code"]
+                        city = data["city"]
+                        country = record.env['res.country'].search([('code', '=', data["country_code"])])
+                        if country:
+                            country_id = country.id
+                            state = record.env['res.country.state'].search([('code', '=', data["state_code"]),('country_id', '=', country_id)])
+                            if state:
+                                state_id = state.id
+                    record.write({
+                        'description': record.description or data['description'],
+                        'partner_name': record.partner_name or data['name'],
+                        'reveal_id': record.reveal_id or data['clearbit_id'],
+                        'website': record.website or ('https://www.%s' % data['domain'] if data['domain'] else False),
+                        'phone': record.phone or (data["phone_numbers"][0] if (len(data["phone_numbers"])>0) else False),
+                        'mobile': record.mobile or (data["phone_numbers"][1] if (len(data["phone_numbers"])>1) else False),
+                        'street': record.street or street,
+                        'city': record.city or city,
+                        'zip': record.zip or zip_code,
+                        'country_id': record.country_id.id or country_id,
+                        'state_id': record.state_id.id or state_id,
+                        'enriched_lead': True,
+                    })
+                    record._send_message(data)
+                else:
+                    record._send_message()
+    @api.model
+    def notify_no_more_credit(self, service_name, model_name, notification_parameter):
```

Comment:
Well I don't know what are the real specifications but I don't think you want to spam every people that once created leads. What are the people to warn ?

Reply:
indeed, I reread the specs, we just need to send a mail to the person in charge of the IAP service. I changed it.